src/dex/maverick-v2/maverick-v2.ts

```diff
@@ -0,0 +1,438 @@
+              // We stop iterating if it becomes 0n at some point
+              let lastOutput = 1n;
+              let dataList: [bigint, number][] = await Promise.all(
+                amounts.map(amount => {
+                  if (amount === 0n) {
+                    return [0n, 0];
+                  }
+                  // We don't want to proceed with calculations if lower amount was not fillable
+                  if (lastOutput === 0n) {
+                    return [0n, 0];
+                  }
+                  const output = pool.swap(
+                    amount,
+                    from,
+                    to,
+                    side == SwapSide.BUY,
+                  );
+                  lastOutput = output[0];
+                  return output;
+                }),
+              );
+
+              let prices = dataList.map(d => d[0]);
+
+              let gasCosts: number[] = dataList.map(
+                ([d, t]: [BigInt, number]) => {
+                  if (d == 0n) return 0;
+                  return MAV_V2_BASE_GAS_COST + MAV_V2_TICK_GAS_COST * t;
+                },
+              );
+
+              return {
+                prices: prices,
+                unit: BigInt(unit),
+                data: {
+                  pool: pool.address,
+                  tokenA: pool.tokenA.address,
+                  tokenB: pool.tokenB.address,
+                  activeTick: state.activeTick,
+                },
+                exchange: this.dexKey,
+                poolIdentifier: pool.name,
+                gasCost: gasCosts,
+                poolAddresses: [pool.address],
+              };
+            } catch (e) {
+              this.logger.debug(
+                `Failed to get prices for pool ${pool.address}, from=${from.address}, to=${to.address}`,
+                e,
+              );
+              return null;
+            }
+          }),
+        )
+      ).filter(isTruthy);
+    } catch (e) {
+      this.logger.error(
+        `Error_getPricesVolume ${srcToken.symbol || srcToken.address}, ${
+          destToken.symbol || destToken.address
+        }, ${side}:`,
+        e,
+      );
+      return null;
+    }
+  }
+
+  // Returns estimated gas cost of calldata for this DEX in multiSwap
+  getCalldataGasCost(
+    poolPrices: PoolPrices<MaverickV2Data>,
+  ): number | number[] {
+    const gasCost = CALLDATA_GAS_COST.DEX_NO_PAYLOAD;
+
+    const arr = new Array(poolPrices.prices.length);
+    poolPrices.prices.forEach((p, index) => {
+      if (p == 0n) {
+        arr[index] = 0;
+      } else {
+        arr[index] = gasCost;
+      }
+    });
+    return arr;
+  }
+
+  // Encode params required by the exchange adapter
+  // Used for multiSwap, buy & megaSwap
+  // Hint: abiCoder.encodeParameter() could be useful
+  getAdapterParam(
+    srcToken: string,
+    destToken: string,
+    srcAmount: string,
+    destAmount: string,
+    data: MaverickV2Data,
+    side: SwapSide,
+  ): AdapterExchangeParam {
```

Comment:
@Wertusser We don't need implementation for this method if we are going to support this integration only on V6